packages/squiggle-lang/src/rescript/Utility/E.res

```diff
@@ -572,12 +572,22 @@ module A = {
       |> (x => Ok(x))
     }
 
-  let getByOpen = (a, op, bin) =>
-    switch getBy(a, r => bin(op(r))) {
```

Comment:
Why go full imperative instead of 
```rescript
let getByOpen = (a, op, bin) =>
    let returned = op(r)
    switch getBy(a, r => bin(returned) {
        Some(r) => Some(returned)
        None => None
    }
```
?

Reply:
You can't, because you don't have ``r`` yet. (I made the same mistake at first)

Reply:
oh lmao, great point. (i have small brain disorder)